src/lib/dapp/DAppSigner.ts

```diff
@@ -106,4 +159,47 @@ export class DAppSigner implements Signer {
   ): Promise<OutputT> {
     throw new Error('Method not implemented.')
   }
+
+  /*
+   *  Extra methods
```

Comment:
I see how these methods could be useful to dApp developers. But exposing these methods may back walletconnect into a corner where it's Signer implementation exposes extra methods that dApps become dependent. I think these extra methods should all be private, unless Signer also publicly exposes the same methods.

Reply:
If you want to use separately the signer functionality from the DAppConnector and session manage it is mandatory to keep those methods public. Furthermore, there are no more issues to having them exposed than having nothing to use.

Reply:
> If you want to use separately the signer functionality from the DAppConnector and session manage it is mandatory to keep those methods public.

I'm not completely understanding this. Can you explain what you mean by `use separately the signer functionality from the DAppConnector and session manage`?

My understanding is that `DAppConnector` should be used to session manage and `DAppSigner` should be used as an implementation for the hashgraph/sdk's Signer interface. Right now `DAppSigner` is leaking its abstraction details to the user.